pkg/api/payer/nodeCursorTracker.go

```diff
@@ -95,15 +95,29 @@ func (ct *NodeCursorTracker) BlockUntilDesiredCursorReached(
 				"Wait for cursor was unsuccessful after %s",
 				time.Since(start),
 			)
-		case err := <-errCh:
+		case err, ok := <-errCh:
```

Comment:
we call defer in the go routine above. Does that mean that this can exit when it should have succeeded?